### PR TITLE
Cache session in getUser

### PR DESCRIPTION
**Pull request by icewind1991**
Querying the server container every time is expensive.

Gives a nice performance increase when working with large folders

[comparison](https://blackfire.io/profiles/compare/65a4966e-f043-4fc9-8cd0-353e122281f0/graph)

cc @DeepDiver1975 @rullzer

### CODE REVIEW

**Comment by rullzer** [comment]
Simple enough. :+1: 
Of course also here I'd like to get rid of the static stuff... but that is for a different pr ;)

**Comment by MorrisJobke** [comment]
Tested, works and code looks good :+1:

**Comment by MorrisJobke** [comment]
:warning: May be superseded by https://github.com/owncloud/core/pull/21419

File: lib/private/user.php [hunk 1]
```diff
@@ -63,7 +63,22 @@ class OC_User {
 	 * @return \OC\User\Session
 	 */
 	public static function getUserSession() {
-		return OC::$server->getUserSession();
+		static $session;
```
Comment (PVince81) [inline lib/private/user.php]: why not just `self::$session` which is more readable, and has the advantage of having a matching declaration in the class?

Comment (icewind1991) [inline lib/private/user.php]: done

**Comment by DeepDiver1975** [comment]
@icewind1991 we will try to patch pimple to get more performance out of the container.

@rullzer volunteered to have a look at pimple.

Do we agree to close this PR?

**Comment by MorrisJobke** [comment]
> Do we agree to close this PR?

Yes

**Pull request closed without merging**